librz/core/p/core_java.c

```diff
@@ -10,1659 +9,321 @@
-			rz_cons_printf("  End PC: (0x%x) 0x%" PFMT64x " 0x%" PFMT64x "\n",
-				exc_entry->end_pc, exc_entry->end_pc + start, exc_entry->file_offset + 2);
-			rz_cons_printf("  Handler PC: (0x%x) 0x%" PFMT64x " 0x%" PFMT64x "\n",
-				exc_entry->handler_pc, exc_entry->handler_pc + start, exc_entry->file_offset + 4);
-			free(class_info);
-		}
-	}
-	return RZ_CMD_STATUS_OK;
-}
-
-RZ_IPI RzCmdStatus rz_cmd_java_handler(RzCore *core, int argc, const char **argv) {
-	if (argc < 2) {
-		return RZ_CMD_STATUS_WRONG_ARGS;
-	}
-	for (int i = 0; i < END_CMDS; i++) {
-		if (!strncmp(argv[1], JAVA_CMDS[i].name, JAVA_CMDS[i].name_len)) {
-			return JAVA_CMDS[i].handler(core, argc - 2, argc > 2 ? &argv[2] : NULL);
-		}
-	}
-	return RZ_CMD_STATUS_WRONG_ARGS;
-}
-
-static const char *cmd_java_subcmd_choices[] = { "help", "set_flags", "prototypes", "resolve_cp", "calc_flags", "flags_str_at", "flags_str", "m_info", "f_info", "find_cp_const", "find_cp_value", "replace_cp_value", "replace_classname_value", "reload_bin", "summary", "exc", "calc_sz", "is_valid", NULL };
-static const RzCmdDescArg cmd_java_args[3] = {
-	{
-		.name = "subcmd",
-		.type = RZ_CMD_ARG_TYPE_CHOICES,
-		.default_value = "help",
-		.choices = cmd_java_subcmd_choices,
-	},
+static const RzCmdDescArg name_args(javar)[] = {
 	{
-		.name = "arg",
-		.type = RZ_CMD_ARG_TYPE_STRING,
-		.flags = RZ_CMD_ARG_FLAG_ARRAY,
-		.optional = true,
+		.name = "index",
+		.type = RZ_CMD_ARG_TYPE_NUM,
 	},
 	{ 0 },
 };
 
-static const RzCmdDescHelp cmd_java_help = {
-	.summary = "Suite of java commands",
-	.description = "Type `java help` for more commands.",
-	.args = cmd_java_args,
+static const RzCmdDescHelp name_help(javar) = {
+	.summary = "resolves the class constant pool value at a given index",
+	.args = name_args(javar),
 };
 
 static bool rz_cmd_java_init_handler(RzCore *core) {
-	RzCmd *cmd = core->rcmd;
-	RzCmdDesc *root_cd = rz_cmd_get_root(cmd);
+	RzCmd *rcmd = core->rcmd;
+	RzCmdDesc *root_cd = rz_cmd_get_root(rcmd);
 	if (!root_cd) {
 		return false;
 	}
 
-	RzCmdDesc *cmd_java_cd = rz_cmd_desc_argv_new(cmd, root_cd, "java", rz_cmd_java_handler, &cmd_java_help);
-	rz_warn_if_fail(cmd_java_cd);
-	return cmd_java_cd != NULL;
+	RzCmdDesc *java = rz_cmd_desc_group_new(rcmd, root_cd, "java", name_handler(java), &name_help(java), &java_usage);
+	if (!java) {
+		rz_warn_if_reached();
+		return false;
+	}
+
+	rz_cmd_desc_argv_modes_new_warn(rcmd, java, javac, RZ_OUTPUT_MODE_STANDARD | RZ_OUTPUT_MODE_JSON);
+	rz_cmd_desc_argv_modes_new_warn(rcmd, java, javaf, RZ_OUTPUT_MODE_STANDARD | RZ_OUTPUT_MODE_JSON);
+	rz_cmd_desc_argv_modes_new_warn(rcmd, java, javai, RZ_OUTPUT_MODE_STANDARD | RZ_OUTPUT_MODE_JSON);
+	rz_cmd_desc_argv_modes_new_warn(rcmd, java, javam, RZ_OUTPUT_MODE_STANDARD | RZ_OUTPUT_MODE_JSON);
+	rz_cmd_desc_argv_modes_new_warn(rcmd, java, javap, RZ_OUTPUT_MODE_STANDARD | RZ_OUTPUT_MODE_JSON);
+	rz_cmd_desc_argv_modes_new_warn(rcmd, java, javas, RZ_OUTPUT_MODE_STANDARD);
```

Comment:
either you fix cmd_api so that STANDANDARD alone == regular command, or you use rz_cmd_desc_argv_new instead, because otherwise:
```
[0x00000339]> javas?
Usage: javas[?]   # prints the class like a java source code
| javas     # prints the class like a java source code
```

Reply:
probably i should.